cosmo_tester/test_suites/snapshots/single_tenant_snapshot_test.py

```diff
@@ -67,7 +68,9 @@ def test_restore_snapshot_and_agents_upgrade_singletenant(
 
     assert_hello_worlds([hello_vm], installed=True, logger=logger)
 
-    check_secrets_converted(new_manager, logger)
+    if old_manager.branch_name in ('3.4.2', '4.0'):
+        # Check we convert secrets for pre-service-user versions
```

Comment:
We don't, no. I think we did briefly but... I'm fairly sure that was changed (leastways I think I was told it was changed, and it might well have been since after 4.0 everything is using service users so it's harder to make things work with keys in arbitrary locations (though I think we should probably still keep on converting them for safety anyway, but... eh...)

I'll enhance the comment on Monday AM.